.github/workflows/ci.yml

```diff
@@ -131,6 +131,28 @@ jobs:
           fail_ci_if_error: false
           verbose: true
 
+  publish:
+    name: Publish crates on release
+    # Only do this job when publishing a release
+    if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
It seems better to do this in a separate workflow with its own [`on: release: types: [published]`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release) trigger.